crates/katana/core/src/service/messaging/starknet.rs

```diff
@@ -61,12 +61,11 @@ impl StarknetMessaging {
         })
     }
 
-    /// Fetches events for the given blocks range.
     pub async fn fetch_events(
         &self,
         from_block: BlockId,
         to_block: BlockId,
-    ) -> Result<HashMap<u64, Vec<EmittedEvent>>> {
+    ) -> Result<Vec<(u64, Vec<EmittedEvent>)>> {
```

Comment:
An `EmittedEvent` already contains the block number. So we may actually simplify even more this function to only return a `Vec<EmittedEvent>` and you can order directly without having to sort.